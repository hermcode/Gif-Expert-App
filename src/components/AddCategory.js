import React, { useState } from 'react'
import PropTypes from 'prop-types'
import 'boxicons'

export const AddCategory = ( {setCategory} ) => {

  const [inputValue, setInputValue] = useState('')

  // Cambia el estado de nuestro input
  const  handleInputChange = (e) => {
    setInputValue(e.target.value);
  }

  const handleSubmit = (e) => {
    e.preventDefault();
    
    if(inputValue.trim().length > 0){
      setCategory( inputValue );
      setInputValue('');
    }
  }

  return (
    <form onSubmit = { handleSubmit } className='search-input'>
      <input 
        type="text"
        value = { inputValue }
        placeholder = "Ingresa una categorÃ­a"
        onChange = { handleInputChange }
      />
      
      <button className="btn-search"><box-icon name='search' size='md' color='white' rotate='90'></box-icon></button>
    </form>
  )
}

AddCategory.propTypes = {
  setCategory: PropTypes.func.isRequired
};