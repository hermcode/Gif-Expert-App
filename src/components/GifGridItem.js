import React from 'react'
import PropTypes from 'prop-types'

export const GifGridItem = ({title, url}) => {
  return (
    <div className = "card animate__animated animate__fadeInUp animate__faster">
      <img src={url} alt={title} />
    </div>
  )
}

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}
