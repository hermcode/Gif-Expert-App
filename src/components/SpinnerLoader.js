import React from 'react'

export const SpinnerLoader = () => {

  const urlImage = 'https://webcorporativa.bankinter.com/www/webcorp/3D/resources/images/precarga2.gif'
  return <img className='spinner-loader' src={ urlImage } alt='Loading'/>  

}
