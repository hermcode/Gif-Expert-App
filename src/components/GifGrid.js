import React, { useEffect, useState } from 'react'
import { GifGridItem } from './GifGridItem';

export const GifGrid = ( {category} ) => {

  const [images, setImages] = useState([]);

  // useEffect
  useEffect( () => {
    getGifs();
  }, [])

  const limitValue = 150;

  // Petición a la API Giphy
  const getGifs = async(limitValue) => {

    const api = 'https://api.giphy.com/v1/gifs/search?'
    const apiKey = '&api_key=wryjtazKZ7YGqtjhFXJkVj1waY1OcmgM';
    const query = `q=${category}`;
    const limit = `&limit=${limitValue}`;
  
    const url = api + query + limit + apiKey;
    const resp = await fetch ( url );
    const {data} = await resp.json();
    
    // Elimina toda la información irrelevante de data
    const gifs = data.map( img => {
      return {
        id: img.id,
        title: img.title,
        url: img.images.downsized_medium.url
      }
    });

    console.log(gifs)
    setImages( gifs );
    
  }

  return (
    <div className='GifGrid'>
      <h2>{ category } <span>{ limitValue } GIFs</span></h2>
      <div className = "card-grid">
        {
          images.map( item =>
            <GifGridItem  
              key = { item.id } 
              title = { item.title }
              url = { item.url }
            />
          )
        }
      </div>
    </div>
  )
}
