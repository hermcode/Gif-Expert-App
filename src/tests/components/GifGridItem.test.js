import React from 'react';
import { shallow } from 'enzyme'
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas en el archivo GifGridItem.js', () => {
  
  const title = 'Un titulo';
  const url = 'https://localhost/algo.jpg';
  const wrapper = shallow ( <GifGridItem title={title} url={url} /> );
  
  test('Debe mostrar el componente correctamente', () => {

    expect( wrapper ).toMatchSnapshot();

  });

  test('debe mostrar la imagen con el src y alt correctamente', () => {

    const img = wrapper.find('img');

    /*  Formas de extraer propiedades

      img.props().src
      img.props('src')

    */
    
    expect(img.props().src).toBe(url);
    expect(img.props().alt).toBe(title);


  })
  
  test('Debe de tener la clase animate__faster', () => {
    
    const div = wrapper.find('div').prop('className').includes('animate__faster');

    expect(div).toBe(true);

  })
  

});

