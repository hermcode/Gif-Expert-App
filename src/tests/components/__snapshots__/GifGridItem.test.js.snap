// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el archivo GifGridItem.js Debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeInUp animate__faster"
>
  <img
    alt="Un titulo"
    src="https://localhost/algo.jpg"
  />
</div>
`;
